lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -56,62 +54,82 @@ sub check {
-        }
+    # Browser with a logged in user
+    my ($user, $reason) = (undef, 'Not authorized');
+    if ($user = $self->current_user) {
+        ($user, $reason) = (undef, 'Bad CSRF token!') unless $self->valid_csrf;
     }
-    else {                                # No session (probably not a browser)
+
+    # No session (probably not a browser)
+    else {
         my $headers = $self->req->headers;
-        my $key     = $headers->header('X-API-Key');
-        my $hash    = $headers->header('X-API-Hash');
-        my $api_key;
-        if ($key) {
-            $log->debug("API key from client: *$key*");
-            $api_key = $self->schema->resultset("ApiKeys")->find({key => $key});
-        }
-        else {
-            $log->debug("No API key from client.");
-            $reason = "no api key";
-        }
-        if ($api_key) {
-            $log->debug(sprintf "Key is for user '%s'", $api_key->user->username);
-            my $msg                = $self->req->url->to_string;
-            my $timestamp          = $headers->header('X-API-Microtime');
-            my $build_tx_timestamp = $headers->header('X-Build-Tx-Time');
-            my $username           = $api_key->user->username;
-            my $request_ip         = $headers->header("x-forwarded-for") || "unknown";
-            if ($self->_valid_hmac($hash, $msg, $build_tx_timestamp, $timestamp, $api_key)) {
-                $user = $api_key->user;
-            }
-            else {
-                my $log_msg
-                  = sprintf "Rejecting authentication for user '%s' with ip '%s'. Valid key '%s', secret '%s'",
-                  $username, $request_ip, $api_key->key, $api_key->secret;
-                if (!_is_timestamp_valid($build_tx_timestamp, $timestamp)) {
-                    $reason = "timestamp mismatch";
-                    $self->app->log->warn($log_msg . ", $reason");
+
+        # Personal access token
+        if (my $userinfo = $self->req->url->to_abs->userinfo) {
+            $reason = 'invalid personal access token';
+            if ($userinfo =~ /^([^:]+):([^:]+):([^:]+)$/) {
+                my ($username, $key, $secret) = ($1, $2, $3);
+                $log->debug(qq{Personal access token for user "$username"});
+                if ($self->is_local_request || $self->req->is_secure) {
+                    my $ip         = $self->tx->remote_address;
+                    my $reject_msg = qq{Rejecting personal access token for user "$username" with ip "$ip"};
+                    if (my $api_key = $self->schema->resultset('ApiKeys')->find({key => $key})) {
+                        my $real_user = $api_key->user;
+                        if ($real_user && secure_compare($real_user->username, $username)) {
+                            if (secure_compare($api_key->secret, $secret)) { ($user, $reason) = ($real_user, undef) }
+                            else                                           { $log->error("$reject_msg, wrong secret") }
```

Comment:
If a user mis-types its secret, should this really be logged as an error? This is not a problem an admin should be concerned about. The same counts for the other errors.

Reply:
I've based this on the API key auth logging. There it's also an error message.

Reply:
Personally, i don't think it's all that useful and would be happy to change it to whatever we can agree on.

Reply:
I guess it makes sense to adapt the existing API key auth logging then.